### ISSUE TITLE
Generate `type`s instead of `interface`s and `enum`s

### ISSUE THREAD

**Issue by gustavo-shigueo** [reinforced/Reinforced.Typings#280]
Is it possible to generate TypeScript ´type` declarations instead of `interface`s? If not can it be added to the library similar to this:

```cs
[TsType]
class Foo {
    int Bar;
}
```

```ts
type Foo = { Bar: number }
```

**Comment by zolakt** [comment]
Enums to types would be a great addition.
Maybe someone has an existing workaround for that?

**Comment by pavel-b-novikov** [comment]
For types support there should be implemented additional synax nodes. Unfortunately, I don't have time currently for such a change :(

Actually it can be done using exising architecture even without modifying the source code. I would appreciate if someone will do that in a way I could integrate it into the product

**Comment by gustavo-shigueo** [comment]
> Enums to types would be a great addition. Maybe someone has an existing workaround for that?

Yes! Having type unions instead of TS enums would be very handy, especially given how bad TS enums are sometimes

**Comment by gustavo-shigueo** [comment]
> For types support there should be implemented additional synax nodes.

At the time I opened the issue I attempted to create an AST node for types but couldn't quite figure out how to properly integrate it with the library

> Unfortunately, I don't have time currently for such a change :(

I understand, maintaining OSS is really difficult, especially with all the other stuff we have to deal with

> Actually it can be done using exising architecture even without modifying the source code. 

How so? Can you give any pointers to how this could be achieved?
